### PR TITLE
Incorporate Lessons into RSS Feed

### PR DESCRIPTION
**Pull request by walshbr**
DO NOT MERGE YET

Testing the feed using netlify as per #825

### CODE REVIEW

**Comment by walshbr** [comment]
As of right now, the RSS feed in this PR grabs the most recent 10 blog posts, the 10 most recent English lessons, and the 10 most recent Spanish lessons. It could be a bit more DRY, but it wouldn't be difficult to add more translations in the future with this refactoring. Is that acceptable behavior for this @acrymble and @mdlincoln? I am not actually too familiar with RSS feeds, so I was reading around to try to see whether they tend to return everything under the sun on the site or rather just the most recent items (seemed to be the latter). Should be ready to merge as long as we're OK with those parameters and would close #825.

Also note that I found two Spanish translations that had the translation dates in an inverted format - Year-Day-Month instead of Year-Month-Day. This PR takes care of those fixes, which are necessary to get the feed to build correctly. Tagging @arojascastro just so that he can be aware that I touched some es files!

**Review by arojascastro — APPROVED** [review]
I am sorry. I thank you for correcting those dates.

**Comment by walshbr** [comment]
No worries at all @arojascastro! Just wanted someone on the Spanish team aware I had mucked about in your files.

File: feed.xml [hunk 1]
```diff
@@ -5,15 +5,15 @@
...
+		<description>{% if site.description %}{{ site.description | xml_escape }}{% endif %}</description>
 		<link>{{ site.url }}</link>
 		<atom:link href="{{ site.url }}/feed.xml" rel="self" type="application/rss+xml" />
 		{% for post in site.posts limit:10 %}
 			<item>
 				<title>{{ post.title | xml_escape }}</title>
 				{% if post.author.name %}
 					<dc:creator>{{ post.author.name | xml_escape }}</dc:creator>
-				{% endif %}        
+				{% endif %}
```
Comment (mdlincoln) [inline feed.xml:16]: Don't worry about it for this PR, but you should set your text editor to not try and automatically trim whitespace from lines.

File: feed.xml [hunk 2]
```diff
@@ -37,7 +37,36 @@
...
+				{% else %}
+								<description>{{ page.excerpt | xml_escape }}</description>
+				{% endif %}
+				<pubDate>{{ page.date | date: "%a, %d %b %Y %H:%M:%S %z" }}</pubDate>
+				<link>{{ site.url }}{{ page.url }}</link>
+				<guid isPermaLink="true">{{ site.url }}{{ page.url }}</guid>
+			</item>
+				{% endfor %}
+
+				{% assign pages = site.pages | where:'layout','lesson'|  where:'lang','es' | sort:'translation_date' | reverse %}
```
Comment (mdlincoln) [inline feed.xml]: I don't think it makes sense to loop the Spanish lessons distinctly from the English lessons _if they're both ending up in the same feed_.

Either we should stream all the lessons and posts together (so finding all lessons without filtering for `lang == 'en'`) or we should create separate RSS feeds for lessons by language (and separate the blog posts, too). If the latter, we should make sure to use https://www.w3schools.com/XML/rss_tag_language.asp

I'm a fan of [KISS](https://en.wikipedia.org/wiki/KISS_principle) as we don't publish with ferocious frequency, and given how few users subscribe to our RSS, I don't think we have the demand for separating them into different streams right now.

**Review by mdlincoln — CHANGES_REQUESTED** [review]
So one of those comments is really more about the bigger question of whether we have separate RSS streams. I vote no, just keep one RSS feed, but make sure lessons of all languages PLUS posts come through.

We can use [RSS category tags](https://www.w3schools.com/XML/rss_tag_category_channel.asp) to divvy them up - that way if any savvy users desperately objects to seeing blog posts, they can configure their RSS reader to do it.

**Comment by walshbr** [comment]
One clarification - would we just want to list all the published lessons/translations in one big feed @mdlincoln? That's pretty easy and doable.

Or do we care about a smaller feed that only lists the most recently published items? I was having trouble with that specificity given the metadata. It seemed as though I couldn't find a good way to grab the most recent lessons that were English _or_ Spanish. In the case of the English lessons that means I am sorting on the "date" field. But for the Spanish lessons that date field isn't really the relevant metadata, as that field stores when the original corresponding _English_ lesson was published. Instead for Spanish I sort on the "translation_date" field, which the English lessons don't have at all. The logic started to feel too complicated for what we were trying to do, though I can think further and put something together if necessary.

**Comment by mdlincoln** [comment]
Ooof, sorry, you're right, I forgot about the mismatched metadata names 😢  Hmm... if we could write our own jekyll plugin then it'd be easy to cross-walk everything, but since we're on GitHub we can't...

What happens if you sort first by date, and then by translation date? Does it kick the lessons up in the queue?

**Comment by mdlincoln** [comment]
*adds yet another item to list of reasons to migrate the building of this site off of GH pages*

File: feed.xml [hunk 3]
```diff
@@ -37,7 +37,36 @@
 				<link>{{ site.url }}{{ page.url }}</link>
 				<guid isPermaLink="true">{{ site.url }}{{ page.url }}</guid>
 			</item>
-          {% endif %}
         {% endfor %}
+
+				{% assign pages = site.pages | where:'layout','lesson'| where: 'lang', 'en' | sort:'date' | reverse %}
+				{% for page in pages limit:10 %}
+			<item>
```
Comment (mdlincoln) [inline feed.xml]: OK - if we can't get one unified for loop due to the date issue, then we can make this a bit more DRY by moving the `<item>` stuff into an `_include` file. The only thing that would need to differ between the includes would be the field to consult for the date... I'm hoping you'd be able to pass that as an [include parameter](https://jekyllrb.com/docs/includes/#passing-parameters-to-includes)?

Comment (mdlincoln) [reply feed.xml]: although i'm not sure if you can pass variable names as parameters -_-

**Comment by walshbr** [comment]
@mdlincoln - "adds yet another item to list of reasons to migrate the building of this site off of GH pages" - I implemented this on my own blog recently, and it was actually pretty painless. Allowed me to keep things on github pages but also use custom plugins. It would change the branching strategy slightly so is not non-trivial, but I've been pretty impressed with it.

File: feed.xml (hunk 3 above)
Comment (walshbr) [reply feed.xml]: I'll take a look at this more and play around! Knowing that I wasn't missing some obvious fix is helpful, and I'll try out these ideas.

**Comment by mdlincoln** [comment]
... but GH pages expressly forbids custom plugins outside of an approved whitelist? What's the strategy you're thinking of?

**Comment by walshbr** [comment]
Oh sorry - the link didn't come through - http://joshfrankel.me/blog/deploying-a-jekyll-blog-to-github-pages-with-custom-plugins-and-travisci/

It involves deploying all your files to a source branch, and then travis builds everything for you with your custom plugins and sends the built files to the appropriate branch to be deployed.

**Comment by mdlincoln** [comment]
Oh, I see. For now, I'd say the new branching structure would be a complete non-starter for our team - we have a hard enough time as is just using the straightforward one! If we did ever want to migrate off, I'd probably prefer a solution like netlify that'll pull our code and build/host, meaning we don't have to change any of our day-to-day editorial practices.

BUT

that is a convo for another day I think. Do you think you can give the `_includes` strategy a try for this one and see how it goes?

**Comment by walshbr** [comment]
Totally re: it not being worth it. 

Check out that commit? Think I was able to use page.lang to put the logic for the date rendering in the include itself. Much more DRY at least. Also realized that I was looping one time too many. So now we're grabbing 10 most recent posts, 10 most recent Eng publications, and 10 most recent translations. 

Tried sorting by date first and then translation date first, but then we have the opposite problem - we'd get only the translations and no recent Eng lessons. Open to other ideas!

**Review by mdlincoln — APPROVED** [review]

**Comment by mdlincoln** [comment]
Looks great @walshbr  - thanks for taking point on this.

**Pull request merged**